### ISSUE TITLE
Action not fails when older submodule is pushed

### ISSUE THREAD

**Issue by aimestereo** [jtmullen/submodule-branch-check-action#8]
Github Action has this as a first line [submodule-branch-check-action](https://github.com/marketplace/actions/submodule-branch-check)

> A github action to ensure that a submodule is **progressing to a child of the previous version (or not changing)** and that the new submodule version is on a specific branch (optional).

I would expect that Action fail PR check if in PR used older submodule version.

I'm not sure if it's documentation problem or an error, but I find it confusing as my test shows opposite:

```
# in submodule
git checkout HEAD~1

# in main repo
git cm "on master but older commit"

# Github Action PASS (expected to fail)
```

In output only last 2 lines are confusing:
```
Run for PR # 2320 of integrate-check-submodule-workflow into develop on TangleInc/TangleBack
Checkout Branch Histories
Full Brach Histories
From TangleInc/TangleBack
 * branch              integrate-check-submodule-workflow -> FETCH_HEAD
From TangleInc/TangleBack
 * branch              develop    -> FETCH_HEAD
submodule.submodules/release_tool.path
M	submodules/release_tool
Switched to branch 'integrate-check-submodule-workflow'
Your branch is up to date with 'origin/integrate-check-submodule-workflow'.
Submodule 'submodules/release_tool' (git@github.com:TangleInc/release-tool.git) registered for path 'submodules/release_tool'
Submodule path 'submodules/release_tool': checked out '9a5ffbcb037e9788559bf6d80b2f687435d107c5'
Switch to submodule at: submodules/release_tool
Switched to branch 'develop'
M	submodules/release_tool
Your branch is up to date with 'origin/develop'.
Submodule path 'submodules/release_tool': checked out '1be4e947292e53c8d1e8638e4a9baa75e482977b'
Submodule submodules/release_tool Changed from: 1be4e947292e53c8d1e8638e4a9baa75e482977b to 9a5ffbcb037e9788559bf6d80b2f687435d107c5

Check if submodule has been changed on integrate-check-submodule-workflow
grep: ...
PASS: Submodule submodules/release_tool has not been changed on branch integrate-check-submodule-workflow
...
# other changed files (not related to submodule)
...
submodules/release_tool: No such file or directory
```

**Comment by jtmullen** [comment]
Hi! Thanks for the issue report - something does seem to be off here. 

First I want to do a quick clarification on this: 
> I would expect that Action fail PR check if in PR used older submodule version.

This action verifies the submodule has not actively been moved backwards, which means it still could be an older version. If the submodule has commits A, B, C, D and was set to B in the repo the action will pass if it is set to B, C, or D. Just as long as it does not backwards move to A. 

----

Regardless, in this case I would not expect to see: `PASS: Submodule submodules/release_tool has not been changed on branch integrate-check-submodule-workflow` if you made any change of the submodule.  

I seem to have just replicated this locally on my end and found how this issue could happen with the way I parse the change list. 

I've pushed a fix and verified it locally, planning to test some more tomorrow - if you can give it a try and let me know if it fixes your issue that would be great. Use the following branch:
`uses: jtmullen/submodule-branch-check-action@change_parse_fix`

**Comment by jtmullen** [comment]
Hi @aimestereo, I believe both this and #7 have been fixed in the latest release (`v0.5.1-beta`). Please let me know if you have further trouble, and thanks for opening them!

FYI you can also have your workflow track the most recent version of this action using the tag `v0-beta`, that will keep it up to date with any fixes/updates until eventually I decide this is good enough for a v1.
